shared/metrics/src/main/java/io/pravega/shared/metrics/DynamicLoggerImpl.java

```diff
@@ -94,15 +95,18 @@ public void onRemoval(RemovalNotification<String, Meter> removal) {
     public void incCounterValue(String name, long delta, String... tags) {
         Exceptions.checkNotNullOrEmpty(name, "name");
         Preconditions.checkNotNull(delta);
+        long traceId = LoggerHelpers.traceEnter(log, "Metrics: incCounterValue", name, delta, tags);
         try {
             MetricsNames.MetricKey keys = metricKey(name, tags);
             Counter counter = countersCache.get(keys.getCacheKey(), new Callable<Counter>() {
                 @Override
                 public Counter call() throws Exception {
+                    LoggerHelpers.traceLeave(log, "Metrics: createCounter", traceId, keys.getRegistryKey(), tags);
```

Comment:
1. The second argument should match the value you pass into the `traceEnter`, otherwise you will never know where something begins and ends.
2. I suggest removing "Metrics: " from the message. That will be logged anyway by the logger.

Reply:
1: It seems overkill to have traceEnter/traceLeave pair for every single line metrics operation. Those operations are one line call, and unlikely to fail (because most of them are 3rd library calls). 
Here we're interested in knowing whether the operation happened, instead of whether the operation succeeded.
2. Fixed.